packages/gestalt/src/Tabs.js

```diff
@@ -4,48 +4,103 @@ import PropTypes from 'prop-types';
 import Box from './Box.js';
 import Flex from './Flex.js';
 import Link from './Link.js';
+import TapArea from './TapArea.js';
 import Text from './Text.js';
 import { type AbstractEventHandler } from './AbstractEventHandler.js';
 
 type OnChangeHandler = AbstractEventHandler<
-  SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+  | SyntheticMouseEvent<HTMLAnchorElement>
+  | SyntheticKeyboardEvent<HTMLAnchorElement>
+  | SyntheticMouseEvent<HTMLDivElement>
+  | SyntheticKeyboardEvent<HTMLDivElement>,
   {| +activeTabIndex: number |},
 >;
 
-function Circle() {
+function Dot() {
   return (
     <Box
       color="red"
       dangerouslySetInlineStyle={{ __style: { marginTop: '1px' } }}
       height={6}
-      marginStart={2}
       rounding="circle"
       width={6}
     />
   );
 }
 
+const UNDERLINE_HEIGHT = 3;
+
+function Underline() {
+  return (
+    <Box
+      color="darkGray"
+      dangerouslySetInlineStyle={{
+        __style: {
+          borderRadius: 1.5,
+        },
+      }}
+      height={UNDERLINE_HEIGHT}
+      width="100%"
+    />
+  );
+}
+
+const COUNT_HEIGHT_PX = 16;
+
+function Count({ count }: {| count: number |}) {
+  const displayCount = count < 100 ? `${count}` : '99+';
+
+  return (
+    <Box
+      color="red"
+      dangerouslySetInlineStyle={{
+        __style: {
+          padding: `0 ${displayCount.length > 1 ? 3 : 0}px`,
+        },
+      }}
+      height={COUNT_HEIGHT_PX}
+      minWidth={COUNT_HEIGHT_PX}
+      rounding="pill"
+    >
+      <Box
+        dangerouslySetInlineStyle={{
+          __style: { padding: '0 0 1px 1px' },
+        }}
+      >
+        <Text align="center" color="white" size="sm" weight="bold">
+          {displayCount}
```

Comment:
with the addition of numbers here, I think we need an indicatorAccessibilityLabel prop to append here. The tricky thing is do we hide the number, and just use the indicatorAccessibilityLabel, or do we append indicatorAccessibilityLabel to the number

indicator={50}, indicatorAccessibilityLabel={"unread notifications"} vs. indicator={50}, indicatorAccessibilityLabel={"50 unread notifications"} 

Right now users just get "Boards for You 50", which is not helpful

Reply:
This would apply to the 'dot' version too. It could be a follow up, but we shouldn't forget about it